src/scipp/html/formatting_datagroup_html.py

```diff
@@ -22,25 +22,29 @@ def _format_shape(var: Union[Variable, DataArray, Dataset, DataGroup]) -> str:
     return f"({shape_list})"
 
 
-def _format_atomic_value(value, maxlen: int = 5) -> str:
-    value_repr = str(value)[:maxlen]
+def _format_atomic_value(value, maxidx: int = 5) -> str:
+    value_repr = str(value)[:maxidx]
     if len(value_repr) < len(str(value)):
         value_repr += "..."
     return value_repr
 
 
-def _format_dictionary_item(name_item: tuple, maxlen: int = 10) -> str:
+def _format_dictionary_item(name_item: tuple, maxidx: int = 10) -> str:
     name, item = name_item
-    name = _format_atomic_value(name, maxlen=maxlen)
-    type_repr = _format_atomic_value(type(item).__name__, maxlen=maxlen)
+    name = _format_atomic_value(name, maxidx=maxidx)
+    type_repr = _format_atomic_value(type(item).__name__, maxidx=maxidx)
     return "(" + ": ".join((name, type_repr)) + ")"
 
 
 def _format_multi_dim_data(var: Union[Variable, DataArray, Dataset, np.ndarray],
                            max_item_number: int = 2) -> str:
-    if len(var.shape) == 0 and (not isinstance(var, Dataset)):
-        return _format_atomic_value(var.value, maxlen=30)
-    elif isinstance(var, Dataset):
+    if isinstance(var, Variable):
+        if len(var.shape) != len(np.array(var.values).shape):
+            return _format_atomic_value(var.values, maxidx=None)
+        elif len(var.shape) == 0:
+            return _format_atomic_value(var.value, maxidx=30)
```

Comment:
Briefer:

```suggestion
        if var.ndim != lvar.values.ndim:
            return _format_atomic_value(var.values, maxidx=None)
        elif var.ndim == 0:
            return _format_atomic_value(var.value, maxidx=30)
```

Also: Converting to a NumPy array may be *expensive*, never do that!

Reply:
Yes...! I applied this suggestion in the code. Thank you...!